packages/runtime-node/src/core-node/ws-node-host.ts

```diff
@@ -13,16 +14,17 @@ export class WsHost extends BaseHost {
     }
 }
 
-/**
- * TODO: handle disconnect
- */
 export class WsServerHost extends BaseHost implements IDisposable {
     private socketToEnvId = new Map<string, { socket: io.Socket; clientID: string }>();
-    private disposed = false;
+    private disposables = new SafeDisposable(WsServerHost.name);
+    dispose = this.disposables.dispose;
+    isDisposed = this.disposables.isDisposed;
 
     constructor(private server: io.Server | io.Namespace) {
         super();
         this.server.on('connection', this.onConnection);
+        this.disposables.add('connection', () => this.server.off('connection', this.onConnection));
```

Comment:
why two hops? it's just removing listeners. I feel that the pattern should NOT be add disposable for each clear. WDYF?